src/CairoLib.sol

```diff
@@ -19,7 +19,7 @@ library CairoLib {
     /// @param functionSelector The function selector of the Cairo contract function to be called.
```

Comment:
As a general comment: you need to keep the memory version of the functions as most of the library is used inside a contract that would do some processing on the input data. 
The code is in contradiction with your PR description as the memory version are removed 
> Backward Compatibility: The original functions with memory parameters are retained to prevent breaking existing integrations.

I think use cases need to be shown: this should to be done in sync with end-to-end tests in the kakarot repo to ensure the usage is relevant and functional. For now, all use case need preprocessing on input hence using `calldata` is not possible.
You can look at how this library is used in the following PRs.

* https://github.com/kkrt-labs/kakarot/pull/1509/files#diff-f1305f3a8c2ecaa0385df3ba54739df0d26d4be8dfcb8ac4aeeac872a26bd0e8
* https://github.com/kkrt-labs/kakarot/pull/1503/files#diff-697c8db4b10c87362c9cd671e703602dde9814b6e8eed2e95e57b7e09fed573b

Reply:
Alright, I will look more into that.